lib/src/date.dart

```diff
@@ -247,25 +253,14 @@ abstract class Date implements Comparable<Date> {
   int compareTo(Date other) {
     if (identical(this, other)) {
       return 0;
-    } else {
-      if (julianDayNumber == other.julianDayNumber &&
-          hour == other.hour &&
-          minute == other.minute &&
-          second == other.second &&
-          millisecond == other.millisecond) {
-        return 0;
-      } else {
-        if (julianDayNumber >= other.julianDayNumber &&
-            hour >= other.hour &&
-            minute >= other.minute &&
-            second >= other.second &&
-            millisecond >= other.millisecond) {
-          return 1;
-        } else {
-          return -1;
-        }
-      }
     }
+    if (julianDayNumber == other.julianDayNumber) {
```

Comment:
It is better to use else ifs here like original code

Reply:
I personally prefer not to use the redundant `else`'s like the most of the official Dart APIs and some other languages.
Sample: https://github.com/dart-lang/sdk/blob/main/sdk/lib/collection/maps.dart
If it's important to you, tell me to change it before the squash.

Reply:
For this specific library a lot of what it is doing is numerical calculations,
So I preferred code being homogenous and control flow being in it's most clear form ...
This is my personal preference :D
Thanks